iOS/Flows/Conversation/Cells/TimeMachineCollectionViewLayout.swift

```diff
@@ -80,17 +89,39 @@ class TimeMachineCollectionViewLayout: UICollectionViewLayout {
         return true
     }
 
-    override func invalidateLayout() {
-        super.invalidateLayout()
+    override func invalidationContext(forBoundsChange newBounds: CGRect)
+    -> UICollectionViewLayoutInvalidationContext {
+
+        let invalidationContext = super.invalidationContext(forBoundsChange: newBounds)
+
+        guard let customInvalidationContext = invalidationContext as? TimeMachineCollectionViewLayoutInvalidationContext else {
+            return invalidationContext
+        }
+
+        // Changing the bounds doesn't affect item z ranges.
+        customInvalidationContext.shouldRecalculateZRanges = false
+
+        return customInvalidationContext
+    }
```

Comment:
If the layout is invalidated due to bounds change, there's no need to re-run the math. 

In contrast, if items are being added/deleted/moved, then all the math should be updated.